cython_lint/cython_lint.py

```diff
@@ -762,8 +762,11 @@ def _main(
     ext: str,
     line_length: int = 88,
     no_pycodestyle: bool = False,
-    ignore: set[str] = set(),
+    ignore: set[str] | None = None,
 ) -> int:
+    if ignore is None:
+        ignore = set()
+    assert ignore is not None  # help mypy
```

Comment:
ignore can't be None here since we always pass a set, so this could be simplified

Reply:
true, it's just `None` in the tests

Reply:
Actually in the test we always pass a set :)

Reply:
in the test you added, yes, but in the others where it's not specified, the default would be `None`

I just get a bit nervous about adding mutables as default (even though in this case it probably wouldn't have caused any issues)

Reply:
> but in the others where it's not specified, the default would be None

indeed, if it wasn't, then CI would error as 100% coverage is enforced once we get into `_main`

Reply:
> in the test you added, yes, but in the others where it's not specified, the default would be None

ah right, I didn't think about that

Reply:
> I just get a bit nervous about adding mutables as default

+1

Reply:
anyway, thanks for your engagement, and please do let me know if you have feature requests that'd be useful in scikit-learn!